### PR TITLE
Topic 0

### PR DESCRIPTION
**Pull request by ritacecheverria**
Excercises 1 and 2

### CODE REVIEW

File: src/Topic0/AbsFactory.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  String type = "tipo";
+  String driver = "com.mysql.jdbc.Driver";
+  String url = "jdbc:mysql://";
+  String pass = "contraseña";
+  String usr = "usuario";
+
+  //Connect according Db Type
+  public String getConection(String type, String driver, String url, String pass, String usr) {
+    String option = type;
+    if (option == "Oracle") {
```
Comment (sebastianscatularo) [inline src/Topic0/AbsFactory.java]: Here you should be able to do something like

```suggestion
   swtich(type) {
       case "Oracle": return new Oracle(driver, url, pass, usr);
       case "Postgre": return new Postgre(driver, url, pass, usr);
       ...
       default:
           throw new IllegalArgumentException("Unable to recognice " + type);
   }
```

File: src/Topic0/AbsFactory.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      Oracle oracled = new Oracle();
+      oracled.ConectDb();
+    } else if (option == "Postgre") {
+      Postgre postgred = new Postgre();
+      postgred.ConectDb();
+    } else if (option == "MySql") {
+      MySql mysqld = new MySql();
+      mysqld.ConectDb();
+    } else {
+      System.out.println("No existe esa conexion");
```
Comment (sebastianscatularo) [inline src/Topic0/AbsFactory.java]: here usually you should throw an exception

File: src/SnDbConect.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+//Singleton Db Connection Ex 1
+
+public class SnDbConect {
+
+  private static Connection conn = null;
+
+  public static Connection getConnection() { //method getConnection creation
+    try {
+      if (conn == null) {
```
Comment (sebastianscatularo) [inline src/SnDbConect.java]: please take a look at the different kind of singletons on Java, this solution is not safe for threads, as first attempt it is fine, but, please take in account the comment, in real envs you are going to deal with threads all the time

**Comment by sebastianscatularo** [comment]
In general, all if fine, just take a look at the class names, trying to be simply an always try to use nouns for class names and adjectives for interfaces, for the interface is a nice to have not a must, but for classes it usually is a must

File: src/SnDbConect.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+//Singleton Db Connection Ex 1
+
+public class SnDbConect {
```
Comment (sebastianscatularo) [inline src/SnDbConect.java]: SnDbConnection is not a common java name

File: src/SnDbConect.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  private static Connection conn = null;
+
+  public static Connection getConnection() { //method getConnection creation
+    try {
+      if (conn == null) {
+        String driver = "com.mysql.jdbc.Driver";
+        String url = "jdbc:mysql://";
+        String pass = "contraseña";
+        String usr = "usuario";
+        Class.forName(driver);
```
Comment (sebastianscatularo) [inline src/SnDbConect.java]: there is no need to use real sql stuffs

**Comment by ritacecheverria** [comment]
Files changed.

**Pull request merged**